src/Agent.Listener/Environment.cs

```diff
@@ -0,0 +1,338 @@
+                if (string.IsNullOrEmpty(toolPath))
+                {
+                    continue;
+                }
+
+                using (var processInvoker = HostContext.CreateService<IProcessInvoker>())
+                {
+                    string toolOutput = string.Empty;
+                    var outputHandler = new EventHandler<DataReceivedEventArgs>((sender, args) =>
+                    {
+                        if (!string.IsNullOrEmpty(args.Data))
+                        {
+                            toolOutput += args.Data;
+                        }
+                    });
+                    processInvoker.OutputDataReceived += outputHandler;
+                    try
+                    {
+                        await processInvoker.ExecuteAsync(
+                                    workingDirectory: string.Empty,
+                                    fileName: toolPath,
+                                    arguments: cap.CommandArgs,
+                                    environment: null,
+                                    cancellationToken: token);
+                        //TODO: should we check if toolOutput is a valid file path? (XPLAT does not check either)
+                        if (!string.IsNullOrEmpty(toolOutput))
+                        {
+                            capabilities.Add(cap.Name, toolOutput);
+                        }
+                    }
+                    catch (OperationCanceledException)
+                    {
+                        throw;
+                    }
+                    catch (Exception ex)
+                    {
+                        Trace.Error(ex);
+                    }
+                    finally
+                    {
+                        processInvoker.OutputDataReceived -= outputHandler;
+                    }
+                }
+
+                token.ThrowIfCancellationRequested();
+            }
+        }
+
+        private void GetFilteredEnvironmentVars(Dictionary<string, string> vars)
+        {
+            IDictionary envVars = System.Environment.GetEnvironmentVariables();
+
+            // Begin with ignoring env vars declared herein
+            var ignoredEnvVariables = new HashSet<string>(_ignoredEnvVariables);
+
+            // Also ignore env vars specified in the 'VSO_AGENT_IGNORE' env var
+            if (envVars.Contains(EnvIgnore))
+            {
+                var additionalIgnoredVars = ((string)envVars[EnvIgnore]).Split(',');
+                foreach (var ignored in additionalIgnoredVars)
+                {
+                    ignoredEnvVariables.Add(ignored.Trim());
+                }
+            }
+
+            // Get filtered env vars
+            foreach (DictionaryEntry envVar in envVars)
+            {
+                string varName = (string)envVar.Key;
+                string varValue = (string)envVar.Value ?? string.Empty;
+                if (!ignoredEnvVariables.Contains(varName) && varValue.Length < 1024)
```

Comment:
Can we add a comment why we're filtering if >= 1024? Is this a server side restriction?